lib/classes/ConfigSchemaHandler/errorMessageBuilder.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+const _ = require('lodash');
+
+const ANY_OF_EVENT_SCHEMA_PATH =
+  '#/properties/functions/patternProperties/%5E%5Ba-zA-Z0-9-_%5D%2B%24/properties/events/items/anyOf';
+
+// matches '#/properties/functions/patternProperties/%5E%5Ba-zA-Z0-9-_%5D%2B%24/properties/events/items/anyOf/123/additionalProperties'
+const ANY_OF_SUB_ERROR_REGEX = /^#\/properties\/functions\/patternProperties\/%5E%5Ba-zA-Z0-9-_%5D%2B%24\/properties\/events\/items\/anyOf\/[\d]+\/additionalProperties$/;
+
+// matches '#/properties/functions/patternProperties/%5E%5Ba-zA-Z0-9-_%5D%2B%24/properties/events/items/anyOf/17/properties/yourPluginEvent/properties/anotherProp/type'
+const DEEPER_ANY_OF_SUB_ERROR_REGEX = /^#\/properties\/functions\/patternProperties\/%5E%5Ba-zA-Z0-9-_%5D%2B%24\/properties\/events\/items\/anyOf\/[\d]+\/properties\/.+$/;
+
+/*
+ * Structure AJV errors to prevent secondary errors related to 'anyOf' condition
+ * are not shown. Those errors are placed in `suberrors` array of the most
+ * relevant error.
+ */
+const structureAjvErrors = ajvErrors => {
+  // If theres is no 'anyOf' error at schemaPath that equals to
+  // then do nothing and return ajvErrors
+  if (!ajvErrors.filter(err => err.schemaPath === ANY_OF_EVENT_SCHEMA_PATH).length) {
+    return ajvErrors;
+  }
+
+  let anyOfErrorIndex;
+  ajvErrors.forEach((ajvError, key) => {
+    if (ajvError.keyword === 'anyOf' && ajvError.schemaPath === ANY_OF_EVENT_SCHEMA_PATH) {
+      anyOfErrorIndex = key;
```

Comment:
I take we want to pick first matching, in that case `findIndex` will serve better

Reply:
I didn't know about that method either, thanks!